Source/FieldSolver/FiniteDifferenceSolver/HybridPICSolveE.cpp

```diff
@@ -238,12 +416,256 @@ void FiniteDifferenceSolver::HybridPICSolveECylindrical (
+
+            // interpolate the B field to a nodal grid
+            auto const Br_interp = Interp(Br, Br_stag, nodal, coarsen, i, j, 0, 0);
+            auto const Bt_interp = Interp(Bt, Bt_stag, nodal, coarsen, i, j, 0, 0);
+            auto const Bz_interp = Interp(Bz, Bz_stag, nodal, coarsen, i, j, 0, 0);
+
+            // calculate enE = (J - Ji) x B
+            enE_nodal(i, j, 0, 0) = (
+                (jt_interp - jit_interp) * Bz_interp
+                - (jz_interp - jiz_interp) * Bt_interp
+            );
+            enE_nodal(i, j, 0, 1) = (
+                (jz_interp - jiz_interp) * Br_interp
+                - (jr_interp - jir_interp) * Bz_interp
+            );
+            enE_nodal(i, j, 0, 2) = (
+                (jr_interp - jir_interp) * Bt_interp
+                - (jt_interp - jit_interp) * Br_interp
+            );
+        });
+
+        if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+        {
+            amrex::Gpu::synchronize();
+            wt = amrex::second() - wt;
+            amrex::HostDevice::Atomic::Add( &(*cost)[mfi.index()], wt);
+        }
+    }
+
+    // Loop through the grids, and over the tiles within each grid again
+    // for the Yee grid calculation of the E field
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+    for ( MFIter mfi(*Efield[0], TilingIfNotGPU()); mfi.isValid(); ++mfi ) {
+        if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+        {
+            amrex::Gpu::synchronize();
+        }
+        Real wt = amrex::second();
+
+        // Extract field data for this grid/tile
+        Array4<Real> const& Er = Efield[0]->array(mfi);
+        Array4<Real> const& Et = Efield[1]->array(mfi);
+        Array4<Real> const& Ez = Efield[2]->array(mfi);
+        Array4<Real const> const& Jr = Jfield[0]->const_array(mfi);
+        Array4<Real const> const& Jt = Jfield[1]->const_array(mfi);
+        Array4<Real const> const& Jz = Jfield[2]->const_array(mfi);
+        Array4<Real const> const& enE = enE_nodal_mf.const_array(mfi);
+        Array4<Real const> const& rho = rhofield->const_array(mfi);
+        Array4<Real> const& Pe = Pefield->array(mfi);
+
+#ifdef AMREX_USE_EB
+        amrex::Array4<amrex::Real> const& lr = edge_lengths[0]->array(mfi);
+        amrex::Array4<amrex::Real> const& lt = edge_lengths[1]->array(mfi);
+        amrex::Array4<amrex::Real> const& lz = edge_lengths[2]->array(mfi);
+#endif
+
+        // Extract stencil coefficients
+        Real const * const AMREX_RESTRICT coefs_r = m_stencil_coefs_r.dataPtr();
+        int const n_coefs_r = m_stencil_coefs_r.size();
+        Real const * const AMREX_RESTRICT coefs_z = m_stencil_coefs_z.dataPtr();
+        int const n_coefs_z = m_stencil_coefs_z.size();
+
+        // Extract cylindrical specific parameters
+        Real const dr = m_dr;
+        // int const nmodes = m_nmodes;
```

Comment:
```suggestion
```

Reply:
I'm hoping we can extend the implementation soon to support m > 0 modes.